client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -56,38 +56,43 @@ const HostingFeatures = () => {
 			? `/hosting-config/${ siteId }`
 			: `/overview/${ siteId }`
 	);
+	const hasRedirectQueryParam = searchParams.has( 'redirect_to' );
 	const hasEnTranslation = useHasEnTranslation();
 
-	const { data: siteTransferData, refetch: refetchSiteTransferData } = useSiteTransferStatusQuery(
-		siteId || undefined
-	);
+	const { data: siteTransferData } = useSiteTransferStatusQuery( siteId || undefined, {
+		refetchInterval: 3000,
+		refetchIntervalInBackground: true,
+	} );
 	// `siteTransferData?.isTransferring` is not a fully reliable indicator by itself, which is why
 	// we also look at `siteTransferData.status`
 	const isTransferInProgress =
 		siteTransferData?.isTransferring || siteTransferData?.status === transferStates.COMPLETED;
 
 	useEffect( () => {
-		if ( ! siteId ) {
-			return;
-		}
-
-		const interval = setInterval( () => {
-			if ( siteTransferData?.status !== transferStates.COMPLETED ) {
-				refetchSiteTransferData();
+		if ( isSiteAtomic && ! isPlanExpired ) {
+			if ( hasRedirectQueryParam ) {
+				window.location.href = redirectUrl.current;
 			} else {
-				clearInterval( interval );
-				dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
+				// The setTimeout is needed when transitioning a site to Atomic.
+				// Once the transition is completed, the user should be redirected to the hosting configuration page.
+				// However, due to a backend delay, immediate redirection may lead to the login page instead.
+				// To avoid this, we introduce a 5-second delay before redirecting.
+				setTimeout( () => {
+					window.location.href = redirectUrl.current;
+				}, 5000 );
 			}
-		}, 3000 );
-
-		return () => clearInterval( interval );
-	}, [ siteId, siteTransferData?.status, refetchSiteTransferData, dispatch ] );
+		}
+	}, [ isSiteAtomic, isPlanExpired, hasRedirectQueryParam ] );
```

Comment:
This logic is difficult to decipher. I understand that we're adapting to a very erratic bug, but it's difficult to understand why we need/don't need a timeout depending on the presence of a `redirect_to` query parameter. Could you elaborate, @nightnei?

Reply:
Previously we had logic for `redirect_to `, so we should keep immediate redirect for that. And if we have URL w/o that query param - we are considering that we are making redirect manually only for the case when a site bacoms Atomic.

Reply:
Would something like this be more readable but still accomplish our goals?

```
	useEffect( () => {
		if ( ! isSiteAtomic || isPlanExpired ) {
			return;
		}

		if ( hasRedirectQueryParam ) {
			// Immediate redirect when explicitly requested via query param
			window.location.href = redirectUrl.current;
			return;
		}

		// Delay redirect for Atomic site transitions
		// This prevents redirection to the login page due to backend delays
		const redirectTimeout = setTimeout( () => {
			window.location.href = redirectUrl.current;
		}, 5000 );

		return () => clearTimeout( redirectTimeout );
	}, [ isSiteAtomic, isPlanExpired, hasRedirectQueryParam ] );
```

I also added a cleanup function to clear the timeout. Does that feel like a good idea? Or am I being super overly cautious?

Reply:
Adding a comment inside the `if ( hasRedirectQueryParam)` statement makes sense. Adding a cleanup function might make sense, though I think it's extremely unlikely that this effect would ever run again after `isSiteAtomic` flips from `false` to `true`. 

I agree that making it so we return early `if ( ! isSiteAtomic || isPlanExpired )` reads a bit nicer, so we might as well do that. I don't know that we need the statement to be `if ( ! isSiteAtomic || ( isSiteAtomic && isPlanExpired ) )`, though. Since the first half of the condition already checks that `isSiteAtomic != false`, it seems redundant to check that `isSiteAtomic == true` in the second part.